### PR TITLE
ENH: add pytest fixtures for testing the file_prefix and directory formatting

### PR DESCRIPTION
**Pull request by awalter-bnl**
This adds some pytest.fixtures that can be used for testing the formatting of the file_prefix and the directory for most suitcase repos.

### CODE REVIEW

File: suitcase/utils/tests/conftest.py [hunk 1]
```diff
@@ -211,3 +211,30 @@ def _example_data_func(ignore=[]):
         return generate_data(plan_type(ignore), ignore=ignore)
 
     return _example_data_func
+
+
+@pytest.fixture(params=['test-', 'scan_{uid}-'],
+                scope='function')
+def file_prefix_list(request):  # noqa
+    '''Returns a list file_prefixes to be formatted for testing.
```
Comment (mrakitin) [inline suitcase/utils/tests/conftest.py]: The description of this function does not match the return type.

File: suitcase/utils/tests/conftest.py [hunk 2]
```diff
@@ -211,3 +211,30 @@ def _example_data_func(ignore=[]):
...
+            pytest.skip()
+        return request.param
+
+    return _file_prefix_list_func
+
+
+@pytest.fixture(params=['', '/test', '/scan_{uid}'],
+                scope='function')
+def directory_list(request):  # noqa
+    '''Returns a list of directories to be formatted for testing.
```
Comment (mrakitin) [inline suitcase/utils/tests/conftest.py]: Same here.

File: suitcase/utils/tests/conftest.py (hunk 1 above)
Comment (awalter-bnl) [reply suitcase/utils/tests/conftest.py]: fixed

File: suitcase/utils/tests/conftest.py (hunk 2 above)
Comment (awalter-bnl) [reply suitcase/utils/tests/conftest.py]: fixed

**Comment by awalter-bnl** [comment]
>  I am not sure if these `_*_list_func()` functions can be pulled outside of those two fixtures and reused?

So I haven't done this just because I think it makes it a bit easier to understand when reading a particular `fixture` to not have to look around for where the 4 line function is defined.Does this make sense @mrakitin

**Comment by mrakitin** [comment]
Yes, I think it's fine for tests for the purpose of readability.

**Pull request merged**